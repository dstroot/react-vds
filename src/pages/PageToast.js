import React, { PureComponent } from 'react';
import ContactForm from 'ContactForm';

// notificatiions
import ReactNotification from 'react-notifications-component';
import 'react-notifications-component/dist/theme.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

class PageToast extends PureComponent {
  componentDidMount() {
    document.title = `Verite Data Science â€¢ Toast`;
    requestAnimationFrame(() => {
      window.scrollTo(0, 0);
    });
  }

  addNotification = event => {
    this.notificationDOMRef.addNotification({
      width: 330,
      content: (
        <div className="notification-custom-success">
          <div className="notification-custom-icon">
            <FontAwesomeIcon icon="check-circle" size="3x" />
          </div>
          <div className="notification-custom-content">
            <p className="notification-title">Thank You</p>
            <p className="notification-message">We will be in touch soon!</p>
          </div>
        </div>
      ),
      type: 'success',
      insert: 'bottom',
      container: 'bottom-right',
      animationIn: ['animated', 'fadeIn'],
      animationOut: ['animated', 'fadeOut'],
      dismiss: { duration: 6000 },
      dismissable: { click: true },
    });
    // remove focus on button
    event.target.blur();
  };

  render() {
    return (
      <div>
        <ContactForm />
        <div className="container mt-5">
          <div className="row">
            <div className="col mt-5">
              <ReactNotification
                ref={input => (this.notificationDOMRef = input)}
              />
              <button
                onClick={this.addNotification}
                className="btn btn-primary"
              >
                Add Notification
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default PageToast;
