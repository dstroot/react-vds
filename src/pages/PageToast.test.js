import React from 'react';
import PageToast from './PageToast';
import renderer from 'react-test-renderer';

test('PageToast renders the toast page', () => {
  const component = renderer.create(<PageToast />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
