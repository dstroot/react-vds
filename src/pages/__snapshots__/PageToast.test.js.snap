// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageToast renders the toast page 1`] = `
<div>
  <div
    className="position-relative overflow-hidden p-2 p-md-5 m-md-3 text-center"
  >
    <div
      className="col-md-7 p-lg-2 mx-auto my-5"
    >
      <h1
        className="display-4 text-primary"
      >
        Get in Touch
      </h1>
      <p
        className="lead"
      >
        Verite Data Science is providing customized solutions for both startups and Fortune 100 companies. Get in touch to see how we can help.
      </p>
      <div
        className="svg-container mx-auto"
      >
        <svg>
          robot.svg
        </svg>
      </div>
      <br />
      <a
        className="btn btn-lg btn-outline-secondary"
        href="/contact"
        onClick={[Function]}
      >
        Contact Us
      </a>
    </div>
  </div>
  <div
    className="container mt-5"
  >
    <div
      className="row"
    >
      <div
        className="col mt-5"
      >
        <div
          className="react-notification-root"
        >
          <div
            className="notification-container-top-left"
          />
          <div
            className="notification-container-top-right"
          />
          <div
            className="notification-container-bottom-left"
          />
          <div
            className="notification-container-bottom-right"
          />
        </div>
        <button
          className="btn btn-primary"
          onClick={[Function]}
        >
          Add Notification
        </button>
      </div>
    </div>
  </div>
</div>
`;
